peps/pep-9999.rst

```diff
@@ -0,0 +1,233 @@
+::
+
+    my_function(param1=param1, param2=param2)
+
+You may instead write:
+::
+
+    my_function(param1=, param2=)
+
+
+Precedence
+==================
+Python already possesses a very similar feature in f-string interpolation where ``f'{x=}'`` is effectively expanded to ``f'x={x}'`` [1]_.
+
+Several modern languages offer similar features during function invocation, sometimes referred to as 'punning'. For example:
+
+* In Ruby,  ``f(x:, y:)`` is syntactic sugar for ``f(x: x, y: y)`` [2]_
+* In ReasonML, ``f(~x, ~y)`` is syntactic sugar for ``f(~x=x, ~y=y)`` [3]_
+* In SystemVerilog, ``(.mult, .mop1, .data);`` is syntactic sugar for ``(.mult(mult), .mop1(mop1),  .data(data));`` [4]_
+
+Outside of function invocation, even more languages have similar features:
+
+* In OCaml, ``let+ x in …`` is syntactic sugar for ``let+ x = x in …`` [5]_
+* In JavaScript, ``{ x, y }`` is syntactic sugar for ``{x: x, y: y}`` [6]_
+* In Rust, ``User { x, y }`` is shorthand for ``User {x: x, y: y}`` [7]_
+
+Applicability
+==================
+We analysed popular Python libraries using `this script <https://gist.github.com/joshuabambrick/a850d0e0050129b9252c748fa06c48b2>`__ to compute:
+
+* Number of keyword arguments were of the form ``f(x=x)`` at invocation
+* Percentage of keyword arguments had the form ``f(x=x)`` at invocation
+* Number of lines of code could be saved by using this syntactic sugar to reduce the need for line wraps
+
+===================================================================== ================ ============== =============== =====================
+Statistic                                                             `cpython <a_>`__ `numpy <b_>`__ `pandas <c_>`__ `scikit-learn <d_>`__
+===================================================================== ================ ============== =============== =====================
+Number of keyword arguments of the form ``f(x=x)`` at invocation      4,225            2,768          13,235          8,342
+Percentage of keyword arguments of the form ``f(x=x)`` at invocation  11.06%           13.17%         17.24%          18.64%
+Lines saved                                                           290              247            935             794
+===================================================================== ================ ============== =============== =====================
+
+.. _a: https://github.com/python/cpython/pull/111423/
+.. _b: https://github.com/numpy/numpy/pull/25021/
+.. _c: https://github.com/pandas-dev/pandas/pull/55744/
+.. _d: https://github.com/scikit-learn/scikit-learn/pull/27680/
+
+Based on this, the ``f(x=x)`` keyword argument pattern is widespread, accounting for 10-20% of all keyword argument uses.
+
+Syntax Selection
+========================
+While this feature has been proposed on numerous occasions with several different forms [8]_ [9]_ [10]_ [11]_ [12]_, [13]_ we have opted to advocate for the ``f(x=)`` form.
+
+Justification
```

Comment:
TODO: may add [poll](https://discuss.python.org/t/syntactic-sugar-to-encourage-use-of-named-arguments/36217/118) feedback here